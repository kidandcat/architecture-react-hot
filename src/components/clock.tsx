import * as React from "react";
import { observer } from "mobx-react";
import { Store } from "../state/store";

@observer
export class Clock extends React.Component<{}, {}> {
  render() {
    return (
      <div style={{ color: Store.color.value }}>{Store.time.split(" ")[4]}</div>
    );
  }
}
