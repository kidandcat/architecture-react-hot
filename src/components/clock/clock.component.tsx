import * as React from "react";
import { selectTime } from "../../State/selectors";
import { selectColor } from "../../State/selectors";
import { createStructuredSelector } from "reselect";
import { tick } from "../../State/actions/time";
import { connect } from "react-redux";

export type ClockProps = {
  time: string;
  color: string;
  tick: Function;
};

const mapStateToProps = createStructuredSelector({
  time: selectTime,
  color: selectColor
});

const mapDispatchToProps = {
  tick
};

@connect(mapStateToProps, mapDispatchToProps)
export class Clock extends React.Component<ClockProps> {
  componentDidMount() {
    this.props.tick();
  }

  render() {
    const { color, time } = this.props;
    return (
      <div style={{ color: color }}>
        <span>Hour:</span> <span>{time}</span>
      </div>
    );
  }
}
