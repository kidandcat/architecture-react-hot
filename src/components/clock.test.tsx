import * as React from "react";
import { shallow } from "enzyme";

import { Clock } from "./clock";

test("<Clock />", () => {
  const wrapper = shallow(<Clock />);
  expect(wrapper).toMatchSnapshot();
});
