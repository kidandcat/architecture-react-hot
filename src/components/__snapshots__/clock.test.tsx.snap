// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Clock />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "13:04:45",
      "style": Object {
        "color": "black",
      },
    },
    "ref": null,
    "rendered": "13:04:45",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "13:04:45",
        "style": Object {
          "color": "black",
        },
      },
      "ref": null,
      "rendered": "13:04:45",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
